addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java

```diff
@@ -268,11 +268,11 @@ else if ( item.isDirectory() )
                         // open the stream here to prevent deletion while waiting for the transfer back to the user to start...
                         InputStream in = item.openInputStream( true, eventMetadata );
                         final Response.ResponseBuilder builder = Response.ok( new TransferStreamingOutput( in ) );
-                        setInfoHeaders( builder, item, sk, path, true, contentController.getContentType( path ),
+                        setInfoHeaders( builder, item, sk, path, false, MediaType.APPLICATION_JSON,
                                         contentController.getHttpMetadata( item ) );
                         response = responseWithBuilder( builder, builderModifier );
                         // generating .http-metadata.json for npm group retrieve to resolve header requirements
-                        if ( eventMetadata.get( STORAGE_PATH ) != null && StoreType.hosted != st )
+                        if ( eventMetadata.get( STORAGE_PATH ) != null && StoreType.group == st )
```

Comment:
So this step is to clean up the http headers found in the upstream server, is that correct?

Reply:
@jdcasey This step is keeping the original http-meta when download, not using npm http-meta generation logic, that's not necessary, just forcing the content-type as json for setInfoHeaders is ok, while tarball should be a stream type response, which I need to fix.